src/components/datagrid/data_grid_types.ts

```diff
@@ -328,3 +332,11 @@ export interface EuiDataGridOnColumnResizeData {
 export type EuiDataGridOnColumnResizeHandler = (
   data: EuiDataGridOnColumnResizeData
 ) => void;
+
+export interface EuiDataGridRowHeightOptions {
+  defaultHeight: number;
+  initialHeights?: Record<
+    number,
+    number | { lineCount?: number; height?: number }
```

Comment:
This definition allows providing competing configurations instead of enumerating only valid shapes. `ExclusiveUnion` is added instead of `number | { lineCount: number } | { height: number }>` so developers are warned if both keys are present.

```suggestion
    number | ExclusiveUnion<{ lineCount: number }, { height: number }>
```

with

```
import { ExclusiveUnion } from '../common';
```

Reply:
done